backend/spindle/views.py

```diff
@@ -38,92 +43,102 @@ def json_response(self) -> JsonResponse:
-                spindle_response.body,
-            )
-            return SpindleResponse(error=SpindleErrorSource.SPINDLE).json_response()
+        if "results" in spindle_response_json and isinstance(spindle_response_json["results"], str):
+            return spindle_response_json['results']
 
-        tex = spindle_response_json["results"]
+        logging.warn(
+            "Spindle response does not contain valid 'results': %s",
+            json.dumps(spindle_response_json)
+        )
 
-        # Return LaTeX code immediately.
-        if mode == "tex":
-            return SpindleResponse(tex=tex).json_response()
-
-        # Forward LaTeX code to LaTeX service.
-        # Return PDF
-        if mode == "pdf":
-            latex_response = http.request(
-                method="POST",
-                url="http://localhost:32769/",
-                body=tex,
-                headers={"Content-Type": "application/tex"},
-            )
-
-            if latex_response.status != 200:
-                logging.warn(
-                    "Received non-200 response from LaTeX server for input %s", tex
-                )
-                return SpindleResponse(
-                    tex=tex, error=SpindleErrorSource.LATEX
-                ).json_response()
-
-            try:
-                pdf = latex_response.data
-            except KeyError:
-                logging.warn(
-                    "LaTeX response does not contain valid 'data': %s", latex_response
-                )
-                return SpindleResponse(error=SpindleErrorSource.LATEX).json_response()
-
-            pdf_base64_string = base64.b64encode(pdf).decode("utf-8")
-
-            return SpindleResponse(tex=tex, pdf=pdf_base64_string).json_response()
-
-        # Compose a link to Overleaf.
-        # quote() is used to escape special characters.
-        if mode == "overleaf":
-            redirect_url = f"https://www.overleaf.com/docs?snip={quote(tex)}"
-            return SpindleResponse(redirect=redirect_url).json_response()
 
-        # Only if the query param is not 'tex', 'pdf' or 'overleaf'.
-        # This should never happen.
-        logging.warn("Received unexpected mode.")
-        return SpindleResponse(error=SpindleErrorSource.GENERAL).json_response()
+    def read_request(self, request) -> Optional[str]:
+        """Read and validate the HTTP request received from the frontend"""
+        request_body = request.body.decode("utf-8")
+
+        try:
+            parsed_json = json.loads(request_body)
+        except json.JSONDecodeError:
+            logging.warn("Input is not JSON parseable: %s", request_body)
+            return
+
+        if not "input" in parsed_json or not isinstance(parsed_json["input"], str):
+            logging.warn("Key input with value of type string not found:", request_body)
+            return
+
+        return parsed_json['input']
+
+    def latex_response(self, tex):
```

Comment:
You may want to add return types to this function signature (and the ones following this one).